integration/sawtooth_integration/tests/test_identity.py

```diff
@@ -0,0 +1,399 @@
+
+        self.walter.set_pubkey_for_role(
+            "allow_all_batchers",
+            "transactor.batch_signer",
+            permit_keys=["*"],
+            deny_keys=[])
+
+        #
+        # transactor.transaction_signer
+        #
+        # From local configuration Carol is denied from XO, but is allowed all
+        # other transactor permissions, Mallory and Chuck are denied from all
+        # transactor permissions.
+
+        self.walter.set_pubkey_for_role(
+            "allow_carol_and_no_others",
+            "transactor.transaction_signer",
+            permit_keys=[self.carol.public_key, self.walter.public_key],
+            deny_keys=[self.alice.public_key, self.bob.public_key,
+                       self.dave.public_key])
+
+        self.assert_able_to_send(self.carol, Families.INTKEY)
+        self.assert_not_able_to_send(self.carol, Families.XO)
+
+        for t, f in [(self.alice, Families.INTKEY),
+                     (self.alice, Families.XO),
+                     (self.bob, Families.INTKEY),
+                     (self.bob, Families.XO),
+                     (self.dave, Families.INTKEY),
+                     (self.dave, Families.XO),
+                     (self.chuck, Families.INTKEY),
+                     (self.chuck, Families.XO),
+                     (self.mallory, Families.INTKEY),
+                     (self.mallory, Families.XO)]:
+            self.assert_not_able_to_send(t, f)
+
+        self.walter.set_pubkey_for_role(
+            "allow_all_transaction_signers",
+            "transactor.transaction_signer",
+            permit_keys=["*"],
+            deny_keys=[])
+
+        #
+        # transactor.transaction_signer.< tp_name > subsection
+        #
+        # From local configuration Dave is denied from being a batch_signer,
+        # Mallory and Chuck are denied being transactors.
+
+        self.walter.set_pubkey_for_role(
+            "deny_alice_from_xo_allow_bob",
+            "transactor.transaction_signer.xo",
+            permit_keys=[self.bob.public_key, self.dave.public_key],
+            deny_keys=[self.alice.public_key])
+
+        self.assert_able_to_send(self.bob, Families.XO)
+
+        for t, f in [(self.alice, Families.XO), (self.chuck, Families.XO),
+                     (self.mallory, Families.XO), (self.dave, Families.XO)]:
+            self.assert_not_able_to_send(t, f)
+
+        self.walter.set_pubkey_for_role(
+            "deny_bob_from_intkey_allow_dave_alice",
+            "transactor.transaction_signer.intkey",
+            permit_keys=[self.alice.public_key, self.dave.public_key],
+            deny_keys=[self.bob.public_key])
+
+        self.assert_able_to_send(self.alice, Families.INTKEY)
+
+        for t, f in [(self.bob, Families.INTKEY), (self.chuck, Families.INTKEY),
+                     (self.mallory, Families.INTKEY), (self.dave, Families.INTKEY)]:
+            self.assert_not_able_to_send(t, f)
+
+    def assert_able_to_send(self, transactor, family):
```

Comment:
How does this assertion fail?

Reply:
Same for `assert_able_to_send_batch`.

Reply:
It fails with an error if the rest api returns a 400 if transactor permissions don't work out.